### PR TITLE
docs+doctest(to): Add tests, examples and documentation to the to-crate

### PR DESCRIPTION
**Pull request by DavidNiehues**

### CODE REVIEW

**Comment by koraa** [comment]
@aminfa needs `cargo fmt`

**Comment by koraa** [comment]
@aminfa And style fixes in the readme (using prettier)

File: to/src/to/to_trait.rs [hunk 1]
```diff
@@ -1,12 +1,45 @@
+//! TODO: Mod docu
```
Comment (aparcar) [inline to/src/to/to_trait.rs:1]: Is adding mod documentation out of the scope for this PR?

Comment (koraa) [reply to/src/to/to_trait.rs:1]: Should be fixed in https://github.com/rosenpass/rosenpass/pull/556

Comment (koraa) [reply to/src/to/to_trait.rs:1]: Lets see if that CI passes, then @DavidNiehues should cherry pick my fixes into this PR.

**Comment by aparcar** [comment]
CI looks happier, @DavidNiehues please cherry pick the commits from @koraa :)

**Review by aparcar — APPROVED** [review]

**Pull request merged**